src/Solvers/krylov.jl

```diff
@@ -0,0 +1,174 @@
+import Krylov
+import Krylov.FloatOrComplex
+
+using KernelAbstractions: @index, @kernel
+
+struct KrylovField{T, F <: AbstractField} <: AbstractVector{T}
+    field::F
+end
+
+function KrylovField(field::F) where F <: AbstractField
+    T = eltype(field)
+    return KrylovField{T,F}(field)
+end
+
+Base.size(kf::KrylovField) = size(kf.field)
+Base.length(kf::KrylovField) = length(kf.field)
+Base.getindex(kf::KrylovField, i::Int) = getindex(kf.field, i)
+
+function Krylov.kdot(n::Int, x::KrylovField{T}, y::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    res = zero(T)
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                res += _x[i,j,k] * _y[i,j,k]
+            end
+        end
+    end
+    return res
+end
+
+function Krylov.knorm(n::Int, x::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    res = zero(T)
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                res += _x[i,j,k]^2
+            end
+        end
+    end
+    return sqrt(res)
+end
+
+function Krylov.kscal!(n::Int, s::T, x::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _x[i,j,k] = s * _x[i,j,k]
+            end
+        end
+    end
+    return x
+end
+
+function Krylov.kaxpy!(n::Int, s::T, x::KrylovField{T}, y::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _y[i,j,k] += s * _x[i,j,k]
+            end
+        end
+    end
+    return y
+end
+
+function Krylov.kaxpby!(n::Int, s::T, x::KrylovField{T}, t::T, y::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _y[i,j,k] = s * _x[i,j,k] + t * _y[i,j,k]
+            end
+        end
+    end
+    return y
+end
+
+function Krylov.kcopy!(n::Int, y::KrylovField{T}, x::KrylovField{T}) where T <: FloatOrComplex
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _y[i,j,k] = _x[i,j,k]
+            end
+        end
+    end
+    return y
+end
```

Comment:
```suggestion
Krylov.kcopy!(n::Int, y::KrylovField, x::KrylovField) = copyto!(y, x)
```

I think similar 1-liners may be defineable for most of the functions